client/src/index.js

```diff
@@ -12,14 +12,65 @@ import "./index.scss";
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
 import rootReducer from "./reducers";
-
+import zIndex from "@material-ui/core/styles/zIndex";
 const store = createStore(rootReducer, applyMiddleware(thunk));
 
-ReactDOM.render(
+const footerStyle = {
+  backgroundColor: "#fff",
+  fontSize: "10px",
+  boxShadow: "0 2px 6px #bababa",
+  textAlign: "center",
+  padding: "20px",
+  position: "fixed",
+  left: "0",
+  bottom: "0",
+  height: "60px",
+  width: "100%",
+  zIndex: "3",
+};
+const links = {
+  color: "silver",
+  padding: "6px",
+  textDecoration: "underline",
+};
+// A function that routes the user to the right place
+// after login
+const onRedirectCallback = (appState) => {
+  history.push(
+    appState && appState.targetUrl
+      ? appState.targetUrl
+      : window.location.pathname,
+  );
+};
+
+const Footer = () => (
+  <footer style={footerStyle}>
```

Comment:
Let's use styled-components for styling as we would like to stay away from inline-styling. It should be a quick fix with the basic syntax like this: 
```
const StyledFooter = styled.footer`
// paste the styles from footerStyle
`;
```
and then to use `StyledFooter` you can replace line 47 with `<StyledFooter></StyledFooter>`

Reply:
I tried this approach first, but it was causing some issue with HOC functions.  I'll try to rectify it👍

Reply:
Yeah please prefer styled components unless you have to pass styles through props...
Also it seems like you might not need a HOC for this? (I assume you mean something like  `withFooter` but if this is only for the full App then we can just put a component in there?)